examples/ui-demo/src/hooks/useModularAccountV2Client.ts

```diff
@@ -52,6 +65,11 @@ export const useModularAccountV2Client = ({
   // Must destructure the inner fields to use as dependencies in the useEffect hook, otherwise the object reference will be compared and cause an infinite render loop
   const { key, entityId, accountAddress } = localKeyOverride ?? {};
 
+  useEffect(() => {
```

Comment:
I needed this because when I switched between modes from 7702 to default, the client wouldn't update when trying to use the erc20 modal

Reply:
how do we get out of this loading state?

Reply:
❓ if you add a `key` to it using the mode (like the `MintCard` and `TransactionsCard` have, does it fix it w/o this?
```ts
      <Erc20SponsorshipCard accountMode={accountMode} />
      <MintCard accountMode={accountMode} key={`mint-card-${accountMode}`} />
      <TransactionsCard
        accountMode={accountMode}
        key={`transactions-card-${accountMode}`}
      />
      <SolanaNftCard />
```